### PR TITLE
Regenerate with ts generator

### PR DESCRIPTION
**Pull request by mtsmfm**
ref: #48

https://github.com/mtsmfm/language_server-protocol-ruby/pull/37 introduced new generator depends on meta model definition.

It seems it doesn't handle `or` kind well.

I'd like to make sure the difference between ts one and new one.

### CODE REVIEW

File: lib/language_server/protocol/constant/document_diagnostic_report_kind.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module LanguageServer
+  module Protocol
+    module Constant
+      #
+      # The document diagnostic report kinds.
+      #
+      module DocumentDiagnosticReportKind
```
Comment (mtsmfm) [inline lib/language_server/protocol/constant/document_diagnostic_report_kind.rb:7]: I think this should be created but there's no definition on meta model

File: lib/language_server/protocol/constant/moniker_kind.rb [hunk 2]
```diff
@@ -1,26 +0,0 @@
-module LanguageServer
-  module Protocol
-    module Constant
-      #
-      # The moniker kind.
-      #
-      # @since 3.16.0
-      #
-      module MonikerKind
```
Comment (mtsmfm) [inline lib/language_server/protocol/constant/moniker_kind.rb:9]: Shouldn't be removed

File: lib/language_server/protocol/constant/position_encoding_kind.rb [hunk 3]
```diff
@@ -2,30 +2,30 @@ module LanguageServer
...
       # A set of predefined position encoding kinds.
       #
-      # @since 3.17.0
-      #
       module PositionEncodingKind
         #
         # Character offsets count UTF-8 code units.
         #
-        UTF8 = "utf-8"
+        U_T_F8 = 'utf-8'
```
Comment (mtsmfm) [inline lib/language_server/protocol/constant/position_encoding_kind.rb:13]: Should be `UTF8`

File: lib/language_server/protocol/interface/text_document_content_change_event.rb [hunk 4]
```diff
@@ -6,10 +6,10 @@ module Interface
       # it is considered to be the full content of the document.
       #
       class TextDocumentContentChangeEvent
-        def initialize(range:, range_length: nil, text:)
+        def initialize(range: nil, range_length: nil, text:)
```
Comment (mtsmfm) [inline lib/language_server/protocol/interface/text_document_content_change_event.rb:9]: https://github.com/mtsmfm/language_server-protocol-ruby/issues/48 can be fixed by this change

Comment (soutaro) [reply lib/language_server/protocol/interface/text_document_content_change_event.rb:9]: Yes, this is the type we want. 👍

File: lib/language_server/protocol/constant/moniker_kind.rb (hunk 2 above)
Comment (mtsmfm) [reply lib/language_server/protocol/constant/moniker_kind.rb:9]: I've noticed TS generator doesn't handle enum

**Comment by paracycle** [comment]
Hey, apologies for missing these in the original PR. I am happy to work on the meta model based implementation to fix these mistakes, if you are still interested in that.

**Comment by mtsmfm** [comment]
@paracycle No worries. Please feel free to send a PR to handle or kind but as you already noticed it seems the meta model lacks some information which exists on TS definition.

**Comment by paracycle** [comment]
> @paracycle No worries. Please feel free to send a PR to handle `or` kind with new generator but as you already noticed it seems the meta model lacks some information which exists on TS definition.

Thanks. I will work on a PR to improve the generation with Ruby code and properly consider `literal` types in `or` kinds (which seems to be what I missed previously).

I also asked about the meta model and what the source of truth for it is. It [seems](https://github.com/microsoft/language-server-protocol/issues/1541#issuecomment-1247903721) the maintained version is the one inside the node repository. If you would like, I can point to the meta model inside that repo to be sure that we are up to date with the spec changes as well.

Anyway, I guess we can discuss this on the PR I open. 

Thanks again for fixing this bug, and apologies for missing it in the first place.

**Pull request merged**